notion-core/src/main/scala/com/ambiata/notion/io/LocationIO.scala

```diff
@@ -48,6 +48,23 @@ case class LocationIO(configuration: Configuration, @transient s3Client: AmazonS
     case h @ HdfsLocation(path)      => Hdfs.isDirectory(new Path(path)).run(configuration)
   }
 
+  def readUTF8(location: Location): ResultTIO[String] = location match {
```

Comment:
Note this implies read's type is `RIO[ReadError \/ String]` or similar. Again, could have a `readOrFail` operation that turns that into a RIO failure if required, _but_ it should use the error data type to make that failure consistent across file systems rather than relying on specific behaviour of underlying implementations.